tello_sim/ursina_adapter.py

```
@@ -304,15 +393,258 @@ def __init__(self):
+        return os.path.join(os.path.dirname(__file__), 'gates.json')
+
+    def load_gate_specs(self) -> list[dict]:
+        """Load the gate layout from gates.json, falling back to the defaults."""
+        path = self._gates_config_path()
+        if os.path.exists(path):
+            try:
+                with open(path) as f:
+                    data = json.load(f)
+                if isinstance(data, list) and data:
+                    return [dict(g) for g in data]
+            except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)